### PR TITLE
Treewide dependency upgrades

### PR DESCRIPTION
**Pull request by msfjarvis**

### CODE REVIEW

**Comment by Tunous** [comment]
That’s a lot of updates here. Did you check whether any of new versions of these libraries introduced changes that could potentially break the app in runtime (or change its behavior in an unwanted way)? Usually these should be mentioned in release notes.

**Comment by msfjarvis** [comment]
> That’s a lot of updates here. Did you check whether any of new versions of these libraries introduced changes that could potentially break the app in runtime (or change its behavior in an unwanted way)? Usually these should be mentioned in release notes.

I went through the release notes and have been extensively using all features of the app since then and haven't run into any crashes or other regressions. If you want we can hold this off for a few days and allow runtime problems to surface, if any.

File: app/src/main/java/me/saket/dank/ui/preferences/PreferenceGroupsScreen.kt [hunk 1]
```diff
@@ -85,7 +85,7 @@ class PreferenceGroupsScreen(context: Context, attrs: AttributeSet) :
   override fun onSaveInstanceState(): Parcelable? {
     val values = Bundle()
 
-    groupChanges.value.ifPresent { group ->
+    groupChanges.value?.ifPresent { group ->
```
Comment (Tunous) [inline app/src/main/java/me/saket/dank/ui/preferences/PreferenceGroupsScreen.kt]: This looks a bit unusual. Did you check whether we could get rid of the use of `Optional` here? The requirement to use null check suggests that it should be possible.

Comment (msfjarvis) [reply app/src/main/java/me/saket/dank/ui/preferences/PreferenceGroupsScreen.kt]: Not yet, I'll get right on that.

Comment (msfjarvis) [reply app/src/main/java/me/saket/dank/ui/preferences/PreferenceGroupsScreen.kt]: https://github.com/Tunous/Dawn/pull/85/files/c36094af4022f19eebbcb2046abc03061587b32d#diff-240fec58b3237ecd66e74db91fb096e9R214 does not accept null values so it's necessary to wrap it with an `Optional` :(

**Comment by msfjarvis** [comment]
> Looks like the latest update broke 2 tests. See Bitrise log for details.

Checking it out

**Comment by saket** [comment]
Powermockito is such a huge pain. The tests should probably be migrated to use Roboelectric instead.

**Comment by Tunous** [comment]
Does this dependabot require any special setup or you can just enable it and it starts working? I'll probably enable it after merging this if that's the case.

**Comment by msfjarvis** [comment]
> Does this dependabot require any special setup or you can just enable it and it starts working? I'll probably enable it after merging this if that's the case.

You just need to grant access to the repo, configure it over at app.dependabot.com with the language used in the repo, and it will generate pull requests automatically.

**Comment by msfjarvis** [comment]
Haven't had any regressions yet with my [staging](https://github.com/msfjarvis/Dawn/commits/staging) branch, I'd say we get this merged soon-ish, unless there's more concerns that I missed? Letting it stew will only increase the size of changes :(

**Comment by Tunous** [comment]
Sure, I also didn’t experience any issues so I’ll try to get this merged tomorrow if I’ll have some free time.

**Pull request merged**